### DISCUSSION TITLE
Why does the `watch` function have a default value?

### DISCUSSION THREAD

**Discussion by guppy0356** [react-hook-form/react-hook-form#12413] [Q&A] [answered]
Why does the watch function have a default value?

```
const [checkbox, name] = watch(['checkbox', 'name'], { checkbox: false, name: 'Joshua' })
```

It seems unnecessary since we can set default values with the defaultValues option in the useForm function.

```
const {
  register,
  handleSubmit,
  watch
} = useForm<Form>({
  defaultValues: {
    checkbox: true, // we can use defaultValues option here
    name: 'sam'
  }
});
```

**Comment by BrendanC23** [comment]
It's only useful if you don't provide the default values when you call `useForm`. In your example, you don't need to include it.

**Comment by guppy0356** [comment]
@BrendanC23 

Thank you for your response. I provide default values in wath only, but I can't see default values in my form 🤔

**Comment by guppy0356** [comment]
This is my sample component. I can't see 'Joshua' in my browser. I think default values in watch function is other meanings. 

```react
import { useForm } from 'react-hook-form';

type Form = {
  name: string,
  checkbox: boolean
}

let renderCount = 0;

function App() {
  renderCount++;

  const {
    register,
    handleSubmit,
    watch
  } = useForm<Form>();

  watch(['checkbox', 'name'], { checkbox: false, name: 'Joshua' })

  return (
    <>
      <p>{renderCount}</p>
      <form onSubmit={handleSubmit((d) => console.log(d))}>
        {/* <input type="checkbox" { ...register("checkbox"), ...object } /> */}
        <input type="checkbox" { ...register("checkbox") } />
        <input {...register("name")} />
        <input type="submit" />
      </form>  
    </>
  )
}

export default App
```

**Reply by BrendanC23** [reply]
The default value used in `watch` will only affect the values returned by `watch`, not the default values displayed in the input. If you want a default value for the form, you can use `value` when calling `register` or provide them when calling `useForm`. 

The default values when calling watch only prevent `undefined` from being returned. See [this CSB](https://codesandbox.io/p/sandbox/react-hook-form-and-dnd-kit-forked-7rxwgl?workspaceId=41f8053c-3c61-4e57-ba77-c5b0b35ebbcb).

> When defaultValue is not defined, the first render of watch will return undefined because it is called before register. It's recommended to provide defaultValues at useForm to avoid this behaviour, but you can set the inline defaultValue as the second argument.

**Comment by guppy0356** [comment]
@BrendanC23 

> The default value used in watch will only affect the values returned by watch, not the default values displayed in the input. 

Okay. I understand. Thank you for your response 😄 By the way, in what kind of program would you use default values? Do you have any knowledge of this?

**Reply by BrendanC23** [reply, marked as answer]
I always provide the default values when calling `useForm()` for any form I am creating. This is the simplest method, and I recommend you try it. By doing this, I've never needed to provide a default value when using `watch`. I think that's only useful if you're code can't handle having a value of `undefined`.
